rte/kernels/mo_rte_solver_kernels.F90

```diff
@@ -947,5 +949,295 @@ pure subroutine apply_BC_0(ncol, nlay, ngpt, top_at_1, flux_dn) bind (C, name="a
+  real(wp), dimension(ncol,nlay,  ngpt), intent(in   ) :: ssa  ! single-scattering albedo,
+  real(wp), dimension(ncol,nlay,  ngpt), intent(in   ) :: g       ! asymmetry parameter []
+  real(wp), dimension(ncol,nlay,  ngpt), intent(in   ) :: lay_source   ! Planck source at layer average temperature [W/m2]
+  real(wp), dimension(ncol,nlay+1,ngpt), intent(in   ) :: lev_source_inc
+                                      ! Planck source at layer edge for radiation in increasing ilay direction [W/m2]
+                                      ! Includes spectral weighting that accounts for state-dependent frequency to g-space mapping
+  real(wp), dimension(ncol,nlay+1,ngpt), intent(in   ) :: lev_source_dec
+                                             ! Planck source at layer edge for radiation in decreasing ilay direction [W/m2]
+  real(wp), dimension(ncol,       ngpt), intent(in   ) :: sfc_emis     ! Surface emissivity      []
+  real(wp), dimension(ncol,       ngpt), intent(in   ) :: sfc_src      ! Surface source function [W/m2]
+  real(wp), dimension(ncol,nlay+1,ngpt), intent(  out) :: flux_up      ! Radiances [W/m2-str]
+  real(wp), dimension(ncol,nlay+1,ngpt), intent(inout) :: flux_dn      ! Top level must contain incident flux boundary condition
+  ! Local variables
+  real(wp), dimension(ncol,nlay+1,ngpt) :: radn_dn, radn_up ! Fluxes per quad angle
+  real(wp), dimension(ncol,       ngpt) :: Ds_ncol
+
+  real(wp), dimension(ncol,nlay,  ngpt) :: tauLoc           ! rescaled Tau
+  real(wp), dimension(ncol,nlay,  ngpt) :: scaling          ! scaling
 
+  integer :: imu, top_level
+  real    :: weight
+  ! Tang rescaling
+  call scaling_1rescl(ncol, nlay, ngpt, tauLoc, scaling, tau, ssa, g) 	
+  ! ------------------------------------
+  !
+  ! For the first angle output arrays store total flux
+  !
+  top_level = MERGE(1, nlay+1, top_at_1)
+  Ds_ncol(:,:) = Ds(1)
+  weight = 2._wp*pi*weights(1)
+  radn_dn(1:ncol, top_level, 1:ngpt)  = flux_dn(1:ncol, top_level, 1:ngpt) / weight
+
+  call lw_solver_1rescl(ncol, nlay, ngpt, &
+                        top_at_1, Ds_ncol, tauLoc, scaling, &
+                        lay_source, lev_source_inc, lev_source_dec, sfc_emis, sfc_src, &
+                        flux_up, flux_dn)
+
+  flux_up = flux_up * weight
+  flux_dn = flux_dn * weight
+
+  do imu = 2, nmus
+    Ds_ncol(:,:) = Ds(imu)
+    weight = 2._wp*pi*weights(imu)
+    radn_dn(1:ncol, top_level, 1:ngpt)  = flux_dn(1:ncol, top_level, 1:ngpt) / weight
+    call lw_solver_1rescl(ncol, nlay, ngpt, &
+                          top_at_1, Ds_ncol, tauLoc, scaling, &
+                          lay_source, lev_source_inc, lev_source_dec, sfc_emis, sfc_src, &
+                          radn_up, radn_dn)
+
+    flux_up(:,:,:) = flux_up(:,:,:) + weight*radn_up(:,:,:)
+    flux_dn(:,:,:) = flux_dn(:,:,:) + weight*radn_dn(:,:,:)
+  end do
+  end subroutine lw_solver_1rescl_GaussQuad
+
+  pure subroutine scaling_1rescl(ncol, nlay, ngpt, tauLoc, scaling, tau, ssa, g)
```

Comment:
Please try to follow the style of the existing code - procedures are set off with long comment lines at the beginning and end

Reply:
fixed

Reply:
Actually, the code has 2 types of style:

1.   in  rte/kernels-openacc/mo_rte_solver_kernels.F90  procedures are set off with long comment lines at the beginning and end
2.    in rte/kernels/mo_rte_solver_kernels.F90 procedures are set off with ! of comment blocks

According to your comment your preference is 1 but many files use option 2. 
So I made my code alignment according to existing file style.